nilearn/_utils/estimator_checks.py

```diff
@@ -559,18 +544,23 @@ def nilearn_check_generator(estimator: BaseEstimator):
         yield (clone(estimator), check_masker_generate_report)
         yield (clone(estimator), check_masker_generate_report_false)
         yield (clone(estimator), check_masker_inverse_transform)
-        yield (clone(estimator), check_masker_transform_resampling)
+        yield (clone(estimator), check_masker_joblib_cache)
         yield (clone(estimator), check_masker_mask_img)
         yield (clone(estimator), check_masker_mask_img_from_imgs)
         yield (clone(estimator), check_masker_no_mask_no_img)
         yield (clone(estimator), check_masker_refit)
         yield (clone(estimator), check_masker_smooth)
+        yield (clone(estimator), check_masker_transform_resampling)
         yield (clone(estimator), check_masker_transformer)
         yield (
             clone(estimator),
             check_masker_transformer_high_variance_confounds,
         )
 
+        if isinstance(estimator, NiftiMasker):
+            # TODO enforce for other maskers
```

Comment:
implement shelving for other maskers in follow up PR?
- to be discussed

Reply:
https://joblib.readthedocs.io/en/latest/memory.html#shelving-using-references-to-cached-values